src/main.rs

```diff
@@ -457,28 +471,29 @@ impl FileReceiver {
 
     /// Write or buffer a chunk that we received for this file.
     fn handle_chunk(&mut self, chunk: Chunk) -> Result<()> {
-        let mut out_file = match self.out_file.take() {
-            None => {
-                let path: &Path = self.fname.as_ref();
-                if let Some(dir) = path.parent() {
-                    std::fs::create_dir_all(dir)?;
-                }
-                let file = File::create(path)?;
-
-                // Resize the file to its final size already:
-                // * So that the file system can do a better job of allocating
-                //   a single extent for it, and it doesn't have to fragment
-                //   the file.
-                // * If we run out of space, we learn about that before we waste
-                //   time on the transfer (although then maybe we should do it
-                //   before we receive a chunk after all?).
-                // This can make debugging a bit harder, because when you look
-                // at just the file size you might think it's fully transferred.
-                file.set_len(self.total_len)?;
-
-                file
+        let path: &Path = self.fname.as_ref();
+        let mut out_file = if !self.file_created {
+            if let Some(dir) = path.parent() {
+                std::fs::create_dir_all(dir)?;
             }
-            Some(f) => f,
+            let file = File::create(path)?;
+
+            // Resize the file to its final size already:
+            // * So that the file system can do a better job of allocating
+            //   a single extent for it, and it doesn't have to fragment
+            //   the file.
+            // * If we run out of space, we learn about that before we waste
+            //   time on the transfer (although then maybe we should do it
+            //   before we receive a chunk after all?).
+            // This can make debugging a bit harder, because when you look
+            // at just the file size you might think it's fully transferred.
```

Comment:
Hm.. this is not resume friendly. If transfer fails in the middle, based on size we wouldn't know transfer was not done. I am not sure is this worth it.

Reply:
I guess it would be fine, if target is temporary location, which will be renamed into place after the transfer. That would be better.